### PR TITLE
Add comments to replay tables

### PR DESCRIPTION
**Pull request by jgdhs27**
Adds comments to replay tables. If the comment contains more than 250 characters, it is cut off and only the first 250 characters are shown. Also adds some small formatting changes to the table, to prevent certain entries (shot name, date) from wrapping and taking up muiltiple lines.

image

### CODE REVIEW

File: project/thscoreboard/shared_content/static/js/replays/replayTable.js [hunk 1]
```diff
@@ -102,6 +102,9 @@ function createTableCell(columnName, value) {
     const text = document.createTextNode(value);
     cell.appendChild(text);
   }
+  if (columnName === "Shot") {
+    cell.style.whiteSpace = "nowrap";
```
Comment (n-rook) [inline project/thscoreboard/shared_content/static/js/replays/replayTable.js:106]: It's easier to understand HTML and make extra changes if we keep as much as possible to CSS. As such, we should put the "white-space: nowrap" in CSS, and here we should add a class to activate that CSS.

File: project/thscoreboard/replays/models.py [hunk 2]
```diff
@@ -360,6 +360,13 @@ def SetForeignKeysFromConstantModels(self, c: ReplayConstantModels):
         self.shot = c.shot
         self.route = c.route
 
+    def GetShortenedComment(self) -> str:
+        """Gets a short version of the replay comment, suitable to be displayed
+        in a table row."""
+        if len(self.comment) <= limits.MAX_SHORTENED_COMMENT_LENGTH:
+            return self.comment
+        return self.comment[:limits.MAX_SHORTENED_COMMENT_LENGTH] + "..."
```
Comment (n-rook) [inline project/thscoreboard/replays/models.py:368]: We need something like this, but in addition I think we should also still use "text-overflow: ellipsis" on the frontend. That'll be good for places like mobile where users likely have small screens.

**Review by n-rook — CHANGES_REQUESTED** [review]

File: project/thscoreboard/shared_content/static/js/replays/replayTable.js (hunk 1 above)
Comment (jgdhs27) [reply project/thscoreboard/shared_content/static/js/replays/replayTable.js]: Added a new style  `.nowrap white-space: nowrap;` to `style.sass`

File: project/thscoreboard/replays/models.py (hunk 2 above)
Comment (jgdhs27) [reply project/thscoreboard/replays/models.py:368]: I actually want the text to be able to take up a couple of lines if needed

Comment (n-rook) [reply project/thscoreboard/replays/models.py:368]: Fair enough! I'm not sure about this, but I don't feel strongly enough to gainsay you about it.

**Review by n-rook — APPROVED** [review]

**Pull request merged**